Test AGP 8.0 using java 17

reidbaker opened the pull request:
- Update Gradle/AGP version and add namespace plus dependencies. 


https://github.com/flutter/flutter/issues/125181

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

reidbaker commented:
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8783181311295663521/+/u/run_test.dart_for_build_tests_shard_and_subshard_1_3/test_stdout
```
Caused by: com.android.builder.errors.EvalIssueException: Android Gradle plugin requires Java 17 to run. You are currently using Java 11."
```

reidbaker committed:
remove more changes to only update namespace

reidbaker committed:
Update AGP in example to minimum requried to support namespace

reidbaker committed:
Merge branch 'master' into i125181-namespace-only

reidbaker committed:
revert all example changes

reidbaker committed:
revert all example changes attempt 2

reidbaker committed:
explicitly set exported on main activity

stuartmorgan commented on packages/integration_test/android/build.gradle:
```diff
@@ -26,7 +26,8 @@ rootProject.allprojects {
 
 android {
-    compileSdkVersion 31
+    namespace = 'dev.flutter.integration_test'
```
I know almost nothing about Groovy, so maybe this is an irrelevant difference, but I've been doing using the syntax `namespace 'dev.flutter.integration_test'`, which is what https://developer.android.com/build/configure-app-module#groovy shows for Groovy.

I see looking around more that https://developer.android.com/build/publish-library/prep-lib-release does the `=`, but maybe there's a difference in how unsupported/unknown things are handled with one syntax vs. the other?

reidbaker replied:
yeah given that groovy is a dynamic language I wonder if the `=` is more strict.  That seems like a reasonable guess.

gmackall replied:
My understanding of groovy is that `foo 'bar'` is equivalent to `foo('bar')` (i.e. a method call), while `foo = 'bar'` is what you would expect it to be. 

But that they end up equivalent in some places because that `foo` is just a setter method for a `foo` property.

See https://stackoverflow.com/questions/22759542/confused-with-the-use-of-assignment-and-brackets (the question is somewhat different but the top answer covers the topic).

reidbaker committed:
remove = from namespace set

reidbaker committed:
Merge branch 'master' into i125181-namespace-only

reidbaker committed:
Update gradle and agp and run updater while keeping package set in mainifest files

reidbaker committed:
Merge branch 'master' into i125181-namespace-only

stuartmorgan commented on packages/integration_test/android/build.gradle:
```diff
@@ -26,7 +26,7 @@ rootProject.allprojects {
 apply plugin: 'com.android.library'
 
 android {
```
You've dropped the `namespace`; without that people still can't use AGP 8.0 in any project that depends on `integration_test`. That's the one change we absolutely *do* need to make for compatibility, based on my testing.

reidbaker replied:
fixed

reidbaker committed:
add namespace back

reidbaker committed:
Conditional namespace

reidbaker committed:
Merge branch 'master' into i125181-namespace-only

stuartmorgan-g reviewed (approved):
LGTM, with one reservation. I'll leave it to your judgement though.

stuartmorgan commented on packages/integration_test/example/android/app/build.gradle:
```diff
@@ -29,6 +29,10 @@ apply plugin: 'com.android.application'
+    // Conditional for compatibility with AGP <4.2.
+    if (project.android.hasProperty("namespace")) {
+        namespace 'com.example.integration_test_example'
+    }
```
Apparently I didn't save the comment that was supposed to go with that overall comment:

Could we make *just* this change in one PR, to make it as safe as possible to cherry-pick, and do the rest in a second PR that we don't cherry-pick? I'm slightly worried that the AGP update here might cause unexpected issues for clients (since IIRC we've hit issues in the past where we thought that wouldn't affect clients and then it did, but maybe I'm misremembering).

reidbaker replied:
https://github.com/flutter/flutter/pull/125657

TahaTesser committed:
Add `slider_perf_android` task to `ci.yaml` (#125540)

This adds `devicelab` task `slider_perf_android` to `.ci.yaml`, which was added in https://github.com/flutter/flutter/pull/125296

jcollins-g committed:
Update dartdoc to v6.2.2. (#125650)

This PR updates the dartdoc version to v6.2.2.

Release notes:  https://github.com/dart-lang/dartdoc/releases/tag/v6.2.2

The main impact for Flutter is to update dartdoc to add chips to class and/or mixin pages for class modifiers.  See dart-lang/dartdoc#3391, dart-lang/dartdoc#3402.   This will also be a cherry-pick candidate for 3.1 if it can land in time.

- [ test-exempt ] I added new tests to check the change I am making, or this PR is [test-exempt].

gmackall committed:
Add migrator to upgrade gradle version when conflict with Android Studio bundled Java version is detected (#124085)

This PR adds an android project migrator that checks the version of android studio and the version of gradle for conflicts, and upgrades to 7.4 if a conflict is detected. For more detail about the particular conflict, see https://github.com/flutter/flutter/issues/122376.

The PR also upgrades older gradle versions being used in integration testing to 7.4. 

Fixes/related to: https://github.com/flutter/flutter/issues/122376 and https://github.com/flutter/flutter/issues/123636

jmagman committed:
Increase Xcode minimum version to 14 (#125639)

Looks like CocoaPods 1.12.1 is incompatible with < Xcode 14 due to https://github.com/CocoaPods/CocoaPods/pull/11828 (see https://github.com/flutter/flutter/issues/123890 for context).

Bump the minimum Xcode version allowed by tooling to 14, released September 2022.

```
[!] Xcode - develop for iOS and macOS (Xcode 13.4)
    â Flutter requires Xcode 14 or higher.
      Download the latest version or update via the Mac App Store.
```

Fixes https://github.com/flutter/flutter/issues/125286.
Previous bump at #97746.

engine-flutter-autoroll committed:
Roll Flutter Engine from 3d61e293cb96 to 1d25906b4cf0 (1 revision) (#125658)

https://github.com/flutter/engine/compare/3d61e293cb96...1d25906b4cf0

2023-04-27 skia-flutter-autoroll@skia.org Roll Skia from f7c8a58c2127 to 05d09f28250a (4 revisions) (flutter/engine#41557)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC jsimmons@google.com,rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

arvinwli committed:
fix RangeSlider, with no overlayShape shifts to the left (#125483)

werainkhatri committed:
enhances OverlayEntry docs (#124478)

Fixes https://github.com/flutter/flutter/issues/123218

fzyzcjy committed:
Allow users to provide route settings to `showMenu` (#124935)

Background: I am adding logging to things like dialogs, bottom sheets and menus. Then I realized that, showMenu does not allow users to provide RouteSettings, while showDialog and showModalBottomSheet both allow. Therefore, IMHO a consistent API design may need to add this to showMenu.

I will add tests if this proposal looks OK :)

reidbaker committed:
update to using agp 8.0 in example

reidbaker committed:
Update lockfiles

reidbaker committed:
Merge branch 'master' into i125181-namespace-only

reidbaker committed:
Update recipes that run integration test to use java 17

reidbaker committed:
Update recipes that run integration test to use java 17

reidbaker committed:
Make test more flexible to changing code error messages

reidbaker committed:
Merge branch 'master' into i125181-namespace-only

reidbaker committed:
Merge branch 'master' into i125181-namespace-only

stuartmorgan commented on packages/flutter_tools/test/integration.shard/android_plugin_compilesdkversion_mismatch_test.dart:
```diff
@@ -23,7 +23,8 @@ void main() {
   });
 
-  test('error logged when plugin Android compileSdkVersion higher than project', () async {
+  test('error logged when plugin Android compileSdkVersion higher than project',
```
It looks like you ran `dart format` on this file, which should be undone. E.g., the breaks after `=` below violate flutter/flutter style (😒).

reidbaker replied:
Undone, I keep forgetting we have code that we are not allowed to auto format :/

stuartmorgan-g reviewed (approved):
LGTM modulo the Dart formatting.

reidbaker committed:
Merge remote-tracking branch 'upstream/master' into i125181-namespace-only

reidbaker committed:
Revert dart format style changes to android_plugin_compilesdkversion_mismatch_test.dart

reidbaker committed:
Update platform_interaction to build with java 17

reidbaker committed:
Update hybrid_android_views to build with java 17 buy updating agp
